SDK/MobileCenter/Microsoft.Azure.Mobile.UWP/Utils/AbstractScreenSizeProvider.cs

```diff
@@ -0,0 +1,26 @@
+﻿using System;
+using System.Threading.Tasks;
+
+namespace Microsoft.Azure.Mobile.Utils
+{
+    public abstract class AbstractScreenSizeProvider : IScreenSizeProvider
+    {
+        // Display height in pixels; -1 indicates unknown
+        public abstract int Height { get; }
+
+        // Display width in pixels; -1 indicates unkown
+        public abstract int Width { get; }
+
+        // Invoked when screen resolution changes (best-effort)
+        public abstract event EventHandler ScreenSizeChanged;
+
+        // Screen size
+        public string ScreenSize
+        {
+            get { return Height == -1 || Width == -1 ? "unknown" : $"{Width}x{Height}"; }
```

Comment:
By now it should be deployed the ingestion change to be optional. Please test to avoid sending "unknown".